Sources/Fuzzilli/Lifting/JavaScriptLifter.swift

```diff
@@ -115,6 +115,44 @@ public class JavaScriptLifter: Lifter {
                 return expr(for: instr.input(idx))
             }
 
+            // Helper function to lift destruct array operations
+            func liftArrayPattern(operation: Operation) -> String {
+                var arrayPattern = ""
+                var outputs: [String] = []
+                var outputIndex = 0
+                var indices: [Int] = []
+                var hasRestElement: Bool = false
+                switch operation {
+                    case let op as DestructArray:
+                        outputs = instr.outputs.map({ $0.identifier })
+                        indices = op.indices
+                        hasRestElement = op.hasRestElement
+                    case let op as DestructArrayAndReassign:
+                        outputs = instr.inputs.dropFirst().map({ $0.identifier })
+                        indices = op.indices
+                        hasRestElement = op.hasRestElement
+                    default:
+                        return arrayPattern
+                }
+                assert(indices.count == outputs.count)
```

Comment:
Can we copy this assertion into the operation constructor as well?

Reply:
I think you mean adding this in ProgramBuilder right? In the Operation constructor we won't have access to the output variables list.

Reply:
Ah yeah right, nevermind. The Operation specifies `numOutput` and that is asserted to be equal to the number of outputs in the Instruction, so we don't need a separate check.